magicgui/tqdm.py

```diff
@@ -0,0 +1,151 @@
+"""A wrapper around the tqdm.tqdm iterator that adds a ProgressBar to a magicgui."""
```

Comment:
Just wondering if there is another place this could live that is not at the top level, i.e. a grouping of a larger thing that it might be an example of? I guess though there isn't anything else quite like in magicgui right now as it's not a widget per se (that is `ProgressBar` itself, this is the iterator) but something that if found inside magic functions with generate a widget?

Reply:
> I guess though there isn't anything else quite like in magicgui right now as it's not a widget per se 

exactly.  this is just the iterator magic.  the main reason I put it at the top level is that I don't want to actually depend on `tqdm` directly (it's available as `magicgui[tqdm]`), and I want people to use `from magicgui.tqdm import tqdm...`.  As you point out, you can still use the manual progressbar API without this (magicgui.widgets.ProgresBar)